File: app_java/src/main/java/vlab/android/architecture/base/BaseViewHolder.java
package vlab.android.architecture.base;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import butterknife.ButterKnife;

/**
 * Created by Vinh Tran on 2/12/18.
 */
public abstract class BaseViewHolder<ItemModel> extends RecyclerView.ViewHolder {
    public BaseViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    abstract protected void bindData(ItemModel item);
}

File: app_kotlin/src/main/java/vlab/android/architecture/base/BaseViewHolder.kt
package vlab.android.architecture.base

import android.support.v7.widget.RecyclerView
import android.view.View

/**
 * Created by Vinh Tran on 2/12/18.
 */
abstract class BaseViewHolder<ItemModel> : RecyclerView.ViewHolder {
    fun getLayoutRes() : Int {
        return 0;
    }
    constructor(itemView: View?) : super(itemView)
    abstract fun bindData(item: ItemModel);
}